lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -158,6 +178,37 @@ def apply_config(config)
 
     private
 
+    sig { params(base_directory: String, excluded_pattern: String).returns(T::Array[String]) }
+    def indexable_included_file_patterns(base_directory = Dir.pwd, excluded_pattern = exclude_pattern)
```

Comment:
Do we need the second argument if it's always just the default?

Reply:
I was trying to avoid computing it again. I suppose I could memoize it into the exclude_pattern method, but I'd need to invalidate that memoization any time users adjust their `@excluded_patterns` list.

Reply:
Also, I'm feeling like `excluded_pattern` isn't as descriptive of a method name as it should be. Rubocop calls it `combined_exclude_glob_patterns`.

Reply:
Also, I realized the code would read more simply if I removed this optional argument and just made a call site pass in `Dir.pwd`.

https://github.com/Shopify/ruby-lsp/pull/2082/commits/b6abcc5ed334feace8150c4ef682f992f14b4272

Reply:
I just added a test to ensure that calling `apply_config` with an excluded_pattern actually does recompute the fnmatch exclude pattern.

https://github.com/Shopify/ruby-lsp/pull/2082/commits/68bb45391d2a2529fa09ecb5e4673abb68179928